sympy/matrices/eigen.py

```diff
@@ -384,8 +384,7 @@ def _is_diagonalizable(M, reals_only=False, dotprodsimp=None, **kwargs):
     if all(e.is_real for e in M) and M.is_symmetric():
         return True
 
-    if all(e.is_complex for e in M) and M.is_hermitian \
-            and not reals_only:
+    if all(e.is_complex for e in M) and M.is_hermitian:

```

Comment:
I'm not sure if this is covered...the `is_diagonalizable` tests are pretty sparse. Something like
```assert Matrix([[0,-4*I],[4*I,0]]).is_diagonalizable()` would work. And for a non-Hermitian case that is diagonalizable in terms of reals, `assert Matrix([[0,-3*I],[4*I,0]]).is_diagonalizable(reals_only=True)`

Reply:
But I'm afraid that such lemmas like
```python
x, y = symbols('x y', real=True)
Matrix([[0, x*I], [y*I, 0]]).is_diagonalizable
```
can be too trivial.

Reply:
At least such tests would provide nominal coverage. I see, however, that codecov indicates that these lines are covered. So once the redundant line is deleted this should be ready to go.